File: Programmers/탐욕법_체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] students = new int[n];
        
        for(int i=0;i<n;i++){
            students[i] = 1;
        }
        
        for(int l:lost){
            students[l-1]--;
        }
        
        for(int r:reserve){
            students[r-1]++;
        }
        
        //빌려주기
        for(int j=0;j<n;j++){
            if(students[j]==0){
                if(j>0 && students[j-1] > 1){
                    students[j-1]--;
                    students[j]++;
                }
                if(j<n-1 && students[j+1] > 1){
                    students[j+1]--;
                    students[j]++;
                }
            }   
        }
        
        //count
        for(int k=0;k<n;k++){
            if(students[k]>0){
                answer++;
            }
        }
        
        return answer;
    }
}

File: Programmers/탐욕법_체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    
    vector<int> student;
    for(int i=0; i<n; i++) student.push_back(1);
    
    for(int l=0; l<lost.size(); l++){
        student[lost[l]-1]--;
    }
    
    for(int r=0; r<reserve.size(); r++){
        student[reserve[r]-1]++;
    }
    
    for(int j=0; j<n; j++){
        //체육복이 없음
        if(student[j] == 0){
            if(student[j-1] > 1 && j-1>=0){
                student[j-1]--;
                student[j]++;
            }
            else if(student[j+1] > 1 && j+1 < n){
                student[j+1]--;
                student[j]++;
            }
        }
    }
    
    answer = n - count(student.begin(), student.end(), 0);
    
    return answer;
}